packages/circuits/circuits/extractor.circom

```diff
@@ -0,0 +1,151 @@
+pragma circom 2.1.6; 
+
+include "circomlib/circuits/comparators.circom";
+include "circomlib/circuits/poseidon.circom";
+
+
+/**
+    return 1 if  left <= element <= right 
+    else return 0;
+**/
+template InRange(n) {
+    signal input left;
+    signal input right; 
+    signal input element; 
+
+    signal output out; 
+
+    component l = GreaterEqThan(n);
+    component r = GreaterEqThan(n);
+
+    l.in[0] <== element; 
+    l.in[1] <== left;
+
+    r.in[0] <== right;
+    r.in[1] <== element;
+
+    out <== l.out * r.out;
+}
+
+
+template PhotoPositionComputation(MAX_NUMBER_BYTES) {
+    signal input dataLen;
+    signal input data[MAX_NUMBER_BYTES];
+    signal input filter[MAX_NUMBER_BYTES];
+
+    signal output photoPosition[2]; 
+
+    signal numberElementLessThan16[MAX_NUMBER_BYTES]; 
+    numberElementLessThan16[0] <== 0;
+    signal lessThan[MAX_NUMBER_BYTES - 1];
+    for (var i = 1; i < MAX_NUMBER_BYTES; i++) {
+        lessThan[i - 1] <== LessThan(8)([filter[i], 16]);
+        numberElementLessThan16[i] <== numberElementLessThan16[i - 1] + lessThan[i  -1 ];
+    }
+    
+    signal totalBasicFieldsSize <== numberElementLessThan16[MAX_NUMBER_BYTES - 1];
+
+    photoPosition[0] <== totalBasicFieldsSize + 1;
+
+    signal index[MAX_NUMBER_BYTES];
+    signal equals[MAX_NUMBER_BYTES];
+    signal acctualLen[MAX_NUMBER_BYTES];
+    signal tmp[MAX_NUMBER_BYTES - 1];
+
+    index[0] <== 0;
+    acctualLen[0] <== 0;
+    equals[0] <== 0;
+
+    for (var i = 1; i < MAX_NUMBER_BYTES; i++) {
+        index[i] <== index[i - 1] + 1;
+        equals[i] <== IsEqual()([index[i], dataLen - 1]);
+        tmp[i - 1] <== data[i - 1] * 256  + data[i];
+        acctualLen[i] <== (tmp[i - 1] - acctualLen[i - 1]) * equals[i] + acctualLen[i - 1];  
+    } 
+
+    photoPosition[1] <== acctualLen[MAX_NUMBER_BYTES - 1]/8 - 65;
+
+}
+
+
+template Extractor(MAX_NUMBER_BYTES) {
+    signal input data[MAX_NUMBER_BYTES];
+    signal input dataLen; 
+
+    signal output photoHash;
+
+    signal output basicIdentityHash; 
+
+    signal output four_digit[4];
+    
+    signal s_data[MAX_NUMBER_BYTES];
+ 
+    component data_is255[MAX_NUMBER_BYTES]; 
+
+    s_data[0] <== 0;
+
+    data_is255[0] = IsEqual();
+    data_is255[0].in[0] <== 255;
+    data_is255[0].in[1] <== data[0]; 
```

Comment:
What's the intended goal here?
Because it can be computed inside the loop since it can be replaced by i = 0
Maybe you wanted to set 1st to 0

Reply:
Applied your suggestion!